oauth2_provider/migrations/0006_alter_application_client_secret.py

```diff
@@ -1,18 +1,17 @@
 from django.db import migrations
-from django.contrib.auth.hashers import identify_hasher, make_password
-import logging
 import oauth2_provider.generators
 import oauth2_provider.models
-
+from oauth2_provider.settings import oauth2_settings
 
 def forwards_func(apps, schema_editor):
     """
     Forward migration touches every application.client_secret which will cause it to be hashed if not already the case.
     """
-    Application = apps.get_model('oauth2_provider', 'application')
-    applications = Application.objects.all()
-    for application in applications:
-        application.save(update_fields=['client_secret'])
+    if oauth2_settings.APPLICATION_MODEL == 'oauth2_provider.Application':
+        Application = apps.get_model('oauth2_provider', 'application')
```

Comment:
I went back and forth on this. My reasoning was that custom models could theoretically diverge enough from the abstract models that any forced migrations that target them could potentially fail. Since custom models have their own migration files anyways, my personal preference would be that all forward migrations would be contained therein. Perhaps a notice for swappable model users that they would need to manually create this migration? What do you think?

Reply:
However, diverging from fields that are updated by the oauth2_provider code would not work. Specifically, if the client_secret is not encrypted then the code will fail to authenticate a client that still has the secret in cleartext. Maybe check the (swapped) model's client_secret field and see if it's class is (a subclass of) `oauth2_provider.models.ClientSecretField` and raise an exception if not, saying they have to first update the swapped model to match as it says in the CHANGELOG:

Add updates to the CHANGELOG to clearly warn swapped model users that they have to take action. 

1. pip install django-oauth-toolkit>=2.0
2. make changes to swapped application model's client_secret using ClientSecretField.
3. makemigrations for the swapped app.
4. Add a  dependency to the swapped app's migration of: `run_before=[('oauth2_provider','0006_alter_application_client_secret.py')]`.
5. migrate

Reply:
hmm of course the swapped model is likely extending the `AbstractApplication` abstract model so maybe they just need to do a makemigrations (e.g. no need to actually edit the swapped application's model).

Reply:
I agree. Dont edit the swapped application model. The app containing the swapped model will do its own migration (automatically happens with makemigrations).

Reply:
@josephabrahams In reviewing this I think I may see why people are having so much trouble with swappable models. 0001 only has:
```
dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL)
    ]
```

It should probably have swappable_dependency defined for all the swappable models... That's a separate issue though!